File: app/src/test/java/com/smlnskgmail/jaman/remotetemperaturecontrol/datasend/data/TestMonitorTarget.java
package com.smlnskgmail.jaman.remotetemperaturecontrol.datasend.data;

import com.smlnskgmail.jaman.remotetemperaturecontrol.monitor.MonitorTarget;
import org.jetbrains.annotations.NotNull;

public class TestMonitorTarget implements MonitorTarget {

    @Override
    public void temperatureAvailable(@NotNull String data) {
        printText(data);
    }

    @Override
    public void temperatureMaximumAvailable(@NotNull String data) {
        printText(data);
    }

    @Override
    public void temperatureMinimumAvailable(@NotNull String data) {
        printText(data);
    }

    @Override
    public void humidityAvailable(@NotNull String data) {
        printText(data);
    }

    @Override
    public void humidityMaximumAvailable(@NotNull String data) {
        printText(data);
    }

    @Override
    public void humidityMinimumAvailable(@NotNull String data) {
        printText(data);
    }

    @Override
    public void resetRequired() {
        printText("");
    }

    private void printText(String text) {
        System.out.print(text);
    }

}

File: app/src/test/java/com/smlnskgmail/jaman/remotetemperaturecontrol/datasend/data/TestMonitorTarget.kt
package com.smlnskgmail.jaman.remotetemperaturecontrol.datasend.data

import com.smlnskgmail.jaman.remotetemperaturecontrol.monitor.support.MonitorTarget

class TestMonitorTarget :
    MonitorTarget {

    override fun temperatureAvailable(data: String) {
        printText(data)
    }

    override fun temperatureMaximumAvailable(data: String) {
        printText(data)
    }

    override fun temperatureMinimumAvailable(data: String) {
        printText(data)
    }

    override fun humidityAvailable(data: String) {
        printText(data)
    }

    override fun humidityMaximumAvailable(data: String) {
        printText(data)
    }

    override fun humidityMinimumAvailable(data: String) {
        printText(data)
    }

    override fun resetRequired() {
        printText("")
    }

    private fun printText(text: String) {
        print(text)
    }

}
